mne/io/nsx/tests/test_nsx.py

```diff
@@ -0,0 +1,135 @@
+    assert raw.info["meas_date"].year == 2023
+    assert raw.info["meas_date"].month == 1
+    assert raw.info["chs"][0]["cal"] == 0.6103515625
+    assert raw.info["chs"][0]["range"] == 0.001
+
+    # check raw_extras
+    for r in raw._raw_extras:
+        assert r["orig_format"] == raw.orig_format
+        assert r["orig_nchan"] == 128
+        assert len(r["timestamp"]) == len(r["nb_data_points"])
+        assert len(r["timestamp"]) == len(r["offset_to_data_block"])
+
+    # Check annotations
+    assert len(raw.annotations) == 0
+
+    raw.load_data()
+    raw_data, times = raw[:]
+    n_channels, n_times = raw_data.shape
+    assert times.shape[0] == n_times
+    orig_data = raw_data / (raw.info["chs"][0]["cal"] * raw.info["chs"][0]["range"])
+    assert np.allclose(sum(orig_data[:, 50] - 10 - np.arange(n_channels)), 76.0)
+
+    assert np.allclose(orig_data[n_channels // 2, :100] - 100, np.arange(100))
+
+    data, times = raw.get_data(start=10, stop=20, return_times=True)
+    assert n_channels, 10 == data.shape
+
+    data, times = raw.get_data(start=0, stop=300, return_times=True)
+
+
+@requires_testing_data
+def test_stim_eog_misc_chs_in_nsx():
+    """Test stim/misc/eog channel assignments."""
+    raw = read_raw_nsx(nsx_22_fname, stim_channel="elec127", eog=["elec126"])
+    assert raw.info["chs"][127]["kind"] == FIFF.FIFFV_STIM_CH
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_EOG_CH
+    raw = read_raw_nsx(nsx_22_fname, stim_channel=["elec127"], eog=["elec126"])
+    assert raw.info["chs"][127]["kind"] == FIFF.FIFFV_STIM_CH
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_EOG_CH
+    raw = read_raw_nsx(nsx_22_fname, stim_channel=127, eog=["elec126"])
+    assert raw.info["chs"][127]["kind"] == FIFF.FIFFV_STIM_CH
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_EOG_CH
+    raw = read_raw_nsx(nsx_22_fname, stim_channel=[127], eog=["elec126"])
+    assert raw.info["chs"][127]["kind"] == FIFF.FIFFV_STIM_CH
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_EOG_CH
+    stims = [ch_info["kind"] == FIFF.FIFFV_STIM_CH for ch_info in raw.info["chs"]]
+    assert np.any(stims)
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_EOG_CH
+    with pytest.raises(ValueError) as excinfo:
+        raw = read_raw_nsx(nsx_22_fname, stim_channel=["elec128", 129], eog=["elec126"])
+    assert "Invalid stim_channel" in str(excinfo.value)
+    with pytest.raises(ValueError) as excinfo:
+        raw = read_raw_nsx(nsx_22_fname, stim_channel=("elec128",), eog=["elec126"])
+    assert "Invalid stim_channel" in str(excinfo.value)
+
+    raw = read_raw_nsx(nsx_22_fname, stim_channel="elec127", misc=["elec126", "elec1"])
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_MISC_CH
+    assert raw.info["chs"][1]["kind"] == FIFF.FIFFV_MISC_CH
+
+
+@requires_testing_data
+def test_nsx_ver_21():
+    """Primary tests for NEURALSG reader."""
+    with pytest.raises(NotImplementedError):
+        read_raw_nsx(nsx_21_fname)
```

Comment:
You should have a test that calls `_test_raw_reader(...)`. If you `git grep _test_raw_reader` you'll see examples for inspiration.

Reply:
Thanks! This helps tremendously.